web/documentserver-example/csharp-mvc/Models/FileUtility.cs

```diff
@@ -35,38 +40,172 @@ public static FileType GetFileType(string fileName)
         {
             var ext = Path.GetExtension(fileName).ToLower();
 
-            if (ExtsDocument.Contains(ext)) return FileType.Word;  // word type for document extensions
-            if (ExtsSpreadsheet.Contains(ext)) return FileType.Cell;  // cell type for spreadsheet extensions
-            if (ExtsPresentation.Contains(ext)) return FileType.Slide;  // slide type for presentation extensions
+            if (FormatManager.DocumentExtensions().Contains(ext)) return FileType.Word;  // word type for document extensions
+            if (FormatManager.SpreadsheetExtensions().Contains(ext)) return FileType.Cell;  // cell type for spreadsheet extensions
+            if (FormatManager.PresentationExtensions().Contains(ext)) return FileType.Slide;  // slide type for presentation extensions
 
             return FileType.Word;  // the default type is word
         }
+    }
+
+    public class Format
+    {
+        public string Name { get; }
+        public FileType Type { get; }
+        public List<string> Actions { get; }
+        public List<string> Convert { get; }
+        public List<string> Mime { get; }
+
+        public Format(string name, FileType type, List<string> actions, List<string> convert, List<string> mime)
+        {
+            Name = name;
+            Type = type;
+            Actions = actions;
+            Convert = convert;
+            Mime = mime;
+        }
+
+        public string Extension()
+        {
+            return "." + Name;
+        }
+    }
+
+    public static class FormatManager
+    {
+        public static List<string> FillableExtensions()
+        {
+            return Fillable()
+                .Select(format => format.Extension())
+                .ToList();
+        }
+
+        public static List<Format> Fillable()
+        {
+            return All()
+                .Where(format => format.Actions.Contains("fill"))
+                .ToList();
+        }
```

Comment:
Do you really have to use 2 functions in FormatManager? Like for example `FillableExtensions()` and `Fillable()`.

Reply:
> Do you really have to use 2 functions in FormatManager? Like for example `FillableExtensions()` and `Fillable()`.

Actually, there's no good reason. I looked at how it was implemented in other examples and did the same. If we need an array of action data in the future, we have a ready method.